### PR TITLE
Add Concourse pipeline for IAM module (tf 0.12)

### PR DESCRIPTION
**Pull request by kopachevsky**
issue #188

### CODE REVIEW

File: infra/terraform/test_fixtures/terraform-google-iam.tf [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    "roles/storage.admin",
+    "roles/pubsub.admin",
+    "roles/cloudkms.admin",
+  ]
+}
+
+resource "google_project" "iam" {
+  provider = "google.phoogle"
+  name = "ci-project-factory"
+  project_id = "ci-project-factory"
```
Comment (Jberlinsky) [inline infra/terraform/test_fixtures/terraform-google-iam.tf]: This needs to be changed

File: infra/terraform/test_fixtures/terraform-google-iam.tf [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    "roles/compute.xpnAdmin",
+    "roles/storage.admin",
+    "roles/pubsub.admin",
+    "roles/cloudkms.admin",
+  ]
+}
+
+resource "google_project" "iam" {
+  provider = "google.phoogle"
+  name = "ci-project-factory"
```
Comment (Jberlinsky) [inline infra/terraform/test_fixtures/terraform-google-iam.tf]: This needs to be changed

File: infra/terraform/test_fixtures/terraform-google-iam.tf [hunk 3]
```diff
@@ -0,0 +1,80 @@
+locals {
+  // Roles required by CI service accounts to run the IAM
+  // integration tests.
+  iam_required_roles = [
+    "roles/resourcemanager.organizationAdmin",
+    "roles/compute.xpnAdmin",
```
Comment (Jberlinsky) [inline infra/terraform/test_fixtures/terraform-google-iam.tf]: This permission isn't mentioned in the [project README](https://github.com/terraform-google-modules/terraform-google-iam). Does this need to be updated?

File: infra/terraform/test_fixtures/terraform-google-iam.tf [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  project_id = "ci-project-factory"
+  folder_id = "${google_folder.phoogle_cloud_foundation_cicd.name}"
+  billing_account = "${module.variables.phoogle_billing_account}"
+}
+
+resource "google_project_services" "iam" {
+  provider = "google.phoogle"
+
+  project = "${google_project.iam.id}"
+  services = [
```
Comment (Jberlinsky) [inline infra/terraform/test_fixtures/terraform-google-iam.tf]: Are we sure that this is the correct set of services to enable?

File: infra/terraform/test_fixtures/terraform-google-iam.tf [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    "iamcredentials.googleapis.com",
+    "oslogin.googleapis.com",
+    "serviceusage.googleapis.com",
+  ]
+}
+
+resource "google_service_account" "iam" {
+  provider = "google.phoogle"
+  project      = "${google_project.iam.id}"
+  account_id   = "ci-project-factory"
```
Comment (Jberlinsky) [inline infra/terraform/test_fixtures/terraform-google-iam.tf]: This needs to be changed

File: infra/terraform/test_fixtures/terraform-google-iam.tf [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  display_name = "ci-project-factory"
```
Comment (Jberlinsky) [inline infra/terraform/test_fixtures/terraform-google-iam.tf]: This needs to be changed

File: infra/terraform/test_fixtures/terraform-google-iam.tf [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+resource "random_id" "iam_github_webhook_token" {
+  byte_length = 20
+}
+
+data "template_file" "iam_github_webhook_url" {
+  template = "https://concourse.infra.cft.tips/api/v1/teams/cft/pipelines/$${pipeline}/resources/pull-request/check/webhook?webhook_token=$${webhook_token}"
+  vars {
+    pipeline = "terraform-google-project-factory"
```
Comment (Jberlinsky) [inline infra/terraform/test_fixtures/terraform-google-iam.tf]: This needs to be changed

**Comment by Jberlinsky** [comment]
It looks like the `terraform-google-project-factory.tf` file was copy-pasted and incompletely find-replaced.

File: infra/terraform/test_fixtures/terraform-google-iam.tf (hunk 3 above)
Comment (kopachevsky) [reply infra/terraform/test_fixtures/terraform-google-iam.tf]: Technically we can say it's mentioned not explicitly, compute.xpnAdmin contains those permissions:

```
Project custom: Add compute.subnetworks.getIamPolicy	and compute.subnetworks..setIamPolicy permissions.
```

File: infra/terraform/test_fixtures/terraform-google-iam.tf (hunk 4 above)
Comment (kopachevsky) [reply infra/terraform/test_fixtures/terraform-google-iam.tf]: Unfortunately - no, I can't be sure this list is sufficient, I just think it's quicker way to get actual list by running pipeline and satisfy missing requirements, other option - do it on theoretical level by reading documentation and making guesses, as long I don't have enough permissions to run tests from my phoogle account

**Comment by ingwarr** [comment]
As I see fixtures made for 0.11 terraform. I've finished updating this module for usage with terraform-0.12, so we need to clarify which version should work at CI.

**Comment by kopachevsky** [comment]
> As I see fixtures made for 0.11 terraform. I've finished updating this module for usage with terraform-0.12, so we need to clarify which version should work at CI.

As Aaron mentioned in chat:

> The Terraform fixture files for new pipelines should be written using Terraform 0.11. The pipelines 
> should be configured to use the Docker images which have Terraform 0.12 as the modules being 
> tested are written with 0.12.

**Comment by kopachevsky** [comment]
> It looks like the `terraform-google-project-factory.tf` file was copy-pasted and incompletely find-replaced.

@Jberlinsky I did fixed most of the copy-paste issues, it appears to be I've created PR againts incorrect branch, had to create new one with review fixes included:  https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/209

I'm closing current one PR

**Pull request closed without merging**